Scripts/CI/README_Metadata_StyleCheck/metadata_style_checker.py

```diff
@@ -0,0 +1,338 @@
+            image_name = re.sub(r'\)', '', image_name)
+            self.images = [image_name]
+        except Exception as err:
+            print(f'Error getting image path from README - {self.readme_path} - {err}.')
+            raise err
+
+    def populate_from_paths(self) -> None:
+        """
+        Populate category name and snippets from a sample's folder.
+        """
+        try:
+            category_folder = get_folder_name_from_path(self.folder_path, -2)
+            self.category = format_category_name_from_folder_name(category_folder)
+        except Exception as err:
+            print(f'Error getting category folder name from path - {self.folder_path} - {err}')
+            raise err
+
+        try:
+            self.snippets = self.get_source_code_paths()
+            self.snippets = sorted(self.snippets, key=str.lower)
+        except Exception as err:
+            print(f"Error parsing paths - {self.folder_name} - {err}.")
+            raise err
+
+    def flush_to_json_string(self) -> str:
+        """
+        Write the metadata to a json string.
+        :return: json string
+        """
+        data = dict()
+
+        data["category"] = self.category
+        data["description"] = self.description
+        data["ignore"] = self.ignore
+        data["images"] = self.images
+        data["keywords"] = self.keywords
+        data["redirect_from"] = self.redirect_from
+        data["relevant_apis"] = self.relevant_apis
+        data["snippets"] = self.snippets
+        data["title"] = self.title
+
+        return json.dumps(data, indent=4, sort_keys=True)
+
+def compare_one_metadata(folder_path: str):
+    """
+    A handy helper function to create 1 sample's metadata by running the script
+    without passing in arguments, and write to a separate json for comparison.
+    The path may look like
+    '~/arcgis-runtime-samples-java/analysis/analyze-hotspots'
+    """
+    single_updater = MetadataCreator(folder_path)
+    try:
+        single_updater.populate_from_readme()
+        single_updater.populate_from_paths()
+
+    except Exception as err:
+        print(f'Error populate failed for - {single_updater.folder_name}.')
+        raise err
+
+    json_path = os.path.join(folder_path, 'README.metadata.json')
+
+    try:
+        json_file = open(json_path, 'r')
+        json_data = json.load(json_file)
+    except Exception as err:
+        print(f'Error reading JSON - {folder_path} = {err}')
+        raise err
+    else:
+        json_file.close()
+
+    # Specific check to ensure redirect_from is not [""] which breaks the website build
+    if json_data['redirect_from'] != [""]:
+        single_updater.redirect_from = json_data['redirect_from']
+    else:
+        raise Exception('Error in metadata: redirect_from cannot be [""]')
+
+    new = single_updater.flush_to_json_string()
+    original = json.dumps(json_data, indent=4, sort_keys=True)
+    if new != original:
```

Comment:
I think if we use the output of this script https://devtopia.esri.com/runtime/common-samples/tree/master/tools/java
as the source of truth, and compare it against the README.metadata.json that is uploaded in the PR, then maybe it'd be easier to run the script and just overwrite whichever comes from the PR? It does feel a bit invasive, but there wouldn't be a way to pass this check if the metadata file isn't identical to the script output anyway? 
At this point we maybe need to consider what/where we need to make changes in case this script is stopping us from merging a metadata file as is?

Reply:
Definitely sounds like a good extra step towards automating - I'll add these to the notes in the issue for the next iteration, thanks 👍